cloudinary_cli/utils/api_utils.py

```diff
@@ -96,12 +98,34 @@ def handle_api_command(
 
     args, kwargs = parse_args_kwargs(func, params[1:]) if len(params) > 1 else ([], {})
 
+    if recursive and cursor is None:
+        raise Exception("Using the `--recursive` flag requires a `--cursor` flag.")
+
     res = func(*args, **{
         **kwargs,
         **{k: v for k, v in optional_parameter},
         **{k: parse_option_value(v) for k, v in optional_parameter_parsed},
     })
 
+    if recursive and \
```

Comment:
Extracting this code to a separate function would improve readability of `handle_api_command` function.